liota/dccs/iotcc.py

```diff
@@ -0,0 +1,248 @@
+                        if json_msg["type"] == "create_or_find_resource_response":
+                            if json_msg["body"]["uuid"] != "null":
+                                log.info("FOUND RESOURCE: {0}".format(json_msg["body"]["uuid"]))
+                                self.reg_entity_id = json_msg["body"]["uuid"]
+                                exit()
+                            else:
+                                log.info("Waiting for resource creation")
+                                time.sleep(5)
+                                self.con.send(
+                                    self.registration(self.con.next_id(), entity_obj.entity_id, entity_obj.name,
+                                                      entity_obj.entity_type))
+                except:
+                    raise
+
+            thread = threading.Thread(target=self.con.run)
+            self.con.on_receive = on_receive_safe
+            thread.daemon = True
+            thread.start()
+            self.con.send(
+                self.registration(self.con.next_id(), entity_obj.entity_id, entity_obj.name, entity_obj.entity_type))
+            thread.join()
+            log.info("Resource Registered {0}".format(entity_obj.name))
+            if entity_obj.parent is not None:
+                self._create_relationship(entity_obj.parent.reg_entity_id, self.reg_entity_id)
+                log.info("Relationship Created")
+            if entity_obj.entity_type == "IoT System":
+                self.store_reg_entity_details(entity_obj, self.reg_entity_id)
+            return entity_obj.register(self, self.reg_entity_id)
+
+    def store_reg_entity_details(self, entity_obj, reg_entity_id):
+        config = ConfigParser.RawConfigParser()
+        fullPath = LiotaConfigPath().get_liota_fullpath()
+        if fullPath != '':
+            try:
+                if config.read(fullPath) != []:
+                    try:
+                        uuid_path = config.get('UUID_PATH', 'uuid_path')
+                        uuid_config = ConfigParser.RawConfigParser()
+                        uuid_config.optionxform = str
+                        uuid_config.add_section('GATEWAY')
+                        uuid_config.set('GATEWAY', 'uuid', reg_entity_id)
+                        uuid_config.set('GATEWAY', 'name', entity_obj.name)
+                        with open(uuid_path, 'w') as configfile:
+                            uuid_config.write(configfile)
+                    except ConfigParser.ParsingError, err:
+                        log.error('Could not open config file')
+                else:
+                    raise IOError('Could not open config file ' + fullPath)
+            except IOError, err:
+                log.error('Could not open config file')
+        else:
+            # missing config file
+            log.warn('liota.conf file missing')
+
+    def publish(self, reg_metric):
+        met_cnt = reg_metric.values.qsize()
```

Comment:
Publish functionality in dcc.py made sense. Before publishing a metric, it should be formatted and then send on the con object should be called. Instead of performing all these conversions in the publish() method, do it inside _format_data() method.